lib/algos/list/Extension.js

```diff
@@ -0,0 +1,72 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+ *  Base class of listing extensions.
+ */
+class Extension {
+    /**
+     * This takes a list of parameters and a logger as the inputs.
+     * Derivatives should have their own format regarding parameters.
+     *
+     * @param {Object} parameters - listing parameter from applications
+     * @param {RequestLogger} logger - the logger
+     * @constructor
+     */
+    constructor(parameters, logger) {
+        // inputs
+        this.parameters = parameters;
+        this.logger = logger;
+        // listing results
+        this.res = undefined;
+        this.keys = 0;
+    }
+
+    /**
+     * Generates listing parameters that metadata can understand from the input
+     * parameters. What metadata can understand: gt, gte, lt, lte, limit, keys,
+     * values, reverse; we use the same set of parameters as levelup's.
+     * Derivatives should have their own conversion of their original listing
+     * parameters into metadata listing parameters.
+     *
+     * @return {object} - listing parameters for metadata
+     */
+    genMDParams() {
+        return {};
+    }
+
+    /**
+     * This function receives a data entry from metadata and decides if it will
+     * include the entry in the listing result or not.
+     *
+     * @param {object} entry - a listing entry from metadata
+     *                         expected format: { key, value }
+     * @return {number} - result of filtering the entry:
```

Comment:
I originally defined the results as positive and negative so that we can declare arbitrary values for specific instructions later. But declaring it now is also OK, why not.